front/src/modules/simulationResult/components/SpaceTimeChart/createChart.ts

```diff
@@ -37,7 +38,9 @@ export default function createChart<T extends number | Date>(
           d3.max(
             train[pos].map(
               (section) =>
-                d3.max(section.map((step: ConsolidatedPosition) => step[keyValues[1]])) as T
+                d3.max(
+                  section.map((step: ConsolidatedPosition) => step[keyValues[1] as 'position']) // dataSimulation is SimulationTrain[] so it’s always position here
```

Comment:
We should rename this file createSpaceTimeChart and rename the function too
```suggestion
                  section.map((step: ConsolidatedPosition) => step.position)